detector/detector/detector_sql.py

```diff
@@ -0,0 +1,247 @@
+ WHERE test_name IN ['web_connectivity']
+ AND msm_failure = 'f'
+ AND measurement_start_time >= %(start_date)s
+ AND measurement_start_time < %(end_date)s
+ AND input IN %(urls)s
+
+ AND input IN 'https://twitter.com/'
+ AND probe_cc = 'RU'
+ AND probe_asn = 12389
+
+ GROUP BY test_name, input, probe_cc, probe_asn
+) AS new
+
+ON
+ new.input = blocking_status.input
+ AND new.probe_cc = blocking_status.probe_cc
+ AND new.probe_asn = blocking_status.probe_asn
+ AND new.test_name = blocking_status.test_name
+) AS x
+"""
+    tau = 0.985
+    mu = 1 - tau
+    urls = sorted(set(u for urls in services.values() for u in urls))
+    d = dict(start_date=start_date, end_date=end_date, tau=tau, mu=mu, urls=urls)
+    query(sql, d)
+
+    if 0 and conf.reprocess and log_example:
+        sql = """SELECT accessible_perc, cnt, status, stability FROM blocking_status FINAL
+        WHERE probe_asn = 135300 and probe_cc = 'MM' AND input = 'https://twitter.com/'
+        """
+        it = list(query(sql)[0])
+        it.append(str(start_date))
+        print(repr(it) + ",")
+
+# unused
+def process_historical_data_nopandas(start_date, end_date, interval, services):
+    """Process past data"""
+    run_date = start_date + interval
+    rebuild_status(click, start_date, run_date, services)
+
+    dates = []
+    while run_date < end_date:
+        dates.append(run_date)
+        run_date += interval
+
+    for run_date in tqdm(dates):
+        run_detection_sql(run_date, run_date + interval, services)
+        changes = extract_changes(run_date)
+        if changes:
+            rebuild_feeds(changes)
+
+    log.debug("Done")
+    log.info(
+        (
+            "blocking_status row count",
+            query("SELECT count() FROM blocking_status FINAL")[0],
+        )
+    )
+
+@metrics.timer("rebuild_status")
+def rebuild_status(click, start_date, end_date, services) -> None:
+    log.info("Truncate blocking_status")
+    sql = "TRUNCATE TABLE blocking_status"
+    query(sql)
+
+    log.info("Truncate blocking_events")
+    sql = "TRUNCATE TABLE blocking_events"
+    query(sql)
+
+    log.info("Fill status")
+    sql = """
+INSERT INTO blocking_status (test_name, input, probe_cc, probe_asn,
+  confirmed_perc, pure_anomaly_perc, accessible_perc, cnt, status)
+  SELECT
+    test_name,
+    input,
+    probe_cc,
+    probe_asn,
+    (countIf(confirmed = 't') * 100) / cnt AS confirmed_perc,
+    ((countIf(anomaly = 't') * 100) / cnt) - confirmed_perc AS pure_anomaly_perc,
+    (countIf(anomaly = 'f') * 100) / cnt AS accessible_perc,
+    count() AS cnt,
+    multiIf(
+        accessible_perc < 70, 'BLOCKED',
+        accessible_perc > 90, 'OK',
+        'UNKNOWN')
+    AS status
+FROM fastpath
+WHERE test_name IN ['web_connectivity']
+AND msm_failure = 'f'
+AND input IN %(urls)s
+AND measurement_start_time >= %(start_date)s
+AND measurement_start_time < %(end_date)s
+
+ AND input IN 'https://twitter.com/'
+ AND probe_cc = 'RU'
+ AND probe_asn = 12389
```

Comment:
Why are `https://twitter.com/`, `RU` and `12389` hardcoded in here?

Reply:
It's used for testing. The _sql file is just a "backup" of unused code at this time.

Reply:
Ah, so I should not be reviewing the code inside of this file? I that this file though has the most extensive documentation compared to `detector.py`. Does this mean the docs in here are in fact no longer accurate?

Reply:
As discussed on Slack the design document is at https://docs.google.com/document/d/1WQ6_ybnPbO_W6Tq-xKuHQslG1dSPr4jUbZ3jQLaMqdw  - detector_sql.py can be safely ignored.

Reply:
If that's the case, can we perhaps delete `detector_sql.py` from the PR for review so that it's a bit less code to review and it's less confusing?

Reply:
Ok, removed.